docs/specs/code/TRANSACTION_MEMO.md

```diff
@@ -0,0 +1,156 @@
+
+Note: API rate limits will apply in a [per user/app basis](https://www.dropbox.com/developers/reference/data-ingress-guide), so it shouldn't be a concern for Yoroi itself.
+
+## User flow
+
+The proposed user flow is the following:
+
+1. The user opens the send transaction dialog.
+2. The memo textarea is closed by default. The user can click *"Add memo"* label and one of the following scenarios occurs:
+   - If the user hasn't synced his external storage account yet, he/she will see the text "In order to add private memos to   transactions you need to link Yoroi to a Dropbox account." and a link to the settings page.
+   		- In the settings page, the user can click *"Connect Yoroi to Dropbox"*.
+   - If the user has already synced his external storage account, the memo textarea will be shown.
+3. If the user clicks *"Add memo"* again:
+   - If the textarea is empty, it will be closed.
+   - If it isn't empty, the user will see a dialog window with the text *"Do you want to discard this memo?"* and two buttons showing "Yes/No".
+     - If the user clicks *"No"*, the dialog closes and the memo textarea remains opened with the same content.
+     - If the user clicks *"Yes"*, the dialog window closes and the textarea become cleared.
+4. If user has entered the memo and makes a transaction, the memo will be saved to local storage and then to external storage. If there is a problem saving to external storage, then the user will see an error message: *"An error occurred while syncing with your Dropbox account. You can try to sync again or discard the memo"*. In that case, one of the following scenarios occurs:
+	- If the user clicks on *"Discard the memo"*, the memo will be removed from local storage.
+	- If the user clicks on *"Try again"* Yoroi will try to upload the file again. The message will remain until the message is successfully uploaded.
+5. When the user goes to his summary of transactions and clicks on a transaction to expand it, he/she will see the transaction memo at the very bottom of the transaction item (if the memo exists). The user will also see an option to remove the memo. If the user clicks it, a dialog message will show up with a message and two buttons showing *"Yes/No"*. The message is: *"Are you sure you want to remove the memo for this transaction? Doing so will erase all copies of it and you will not be able to restore it in the future"*.
+6. When the user goes to the "Connect Yoroi to Dropbox" subsection in the settings page, he/she will see a button to connect his/her account if it's not connected already. 
+7. When the user restores a wallet, he/she will be shown a message: *"If you previously connected your Dropbox account to use transactions memos, then you can restore them by going to Settings -> Connect Yoroi to Dropbox"*
+
```

Comment:
We also need a button on the settings screen to reveal the encryption password for the memos

Reply:
I added it in the user-flow section and as a note in the encryption section.